### ISSUE TITLE
[BUG] Text selection after OCR missing last letter

### ISSUE THREAD

**Issue by r2ixuz** [paperless-ngx/paperless-ngx#4493]
### Description

Please see the bug https://github.com/paperless-ngx/paperless-ngx/issues/1309

I just stumbled upon Stirling PDF Docker Container. OCRed documents with this have not the problem when selecting text.

So I tested some things and the reason why this happens in Paperless is that ocrmypdf is using the sandwich renderer (by default I guess). Stirling PDF does use explicitly "--pdf-renderer hocr".

Is there a reason why we don't use this in Paperless?

### Steps to reproduce

Use latest Paperless-ngx via Docker

### Webserver logs

```bash
does not help to the issue
```

### Paperless-ngx version

1.17.4

### Host OS

gentoo x86_64

### Installation method

Docker - official image

**Comment by shamoon** [comment]
The answer to this is the same as your own 1y/o issue which is that when tesseract fix this we can update. The same issue is still open over there: https://github.com/tesseract-ocr/tesseract/issues/2879

As you can see, it doesnt seem to be a simple fix

**Comment by r2ixuz** [comment]
Yes, but this issue is only present when using the sandwich renderer. I just wanted to know why we are using this in Paperless as it is not needed when you can use hocr instead...

I mean literally every OCRed document by Paperless got this buggy text selection behaviour. And this is nothing which comes from the PDF client neither the client OS is relevant.

I don't understand the mentality in this project to fingerpoint every single time to upstream issues. You can decide which version of upstream software you use and how do you configure them. Just updating the upstream packages on and on is not enough IMHO.

**Comment by stumpylog** [comment]
The default pdf render option in OCRMyPdf is "atuo", which, per their documentation, ["always selects `sandwich`"](https://ocrmypdf.readthedocs.io/en/latest/advanced.html#changing-the-pdf-renderer).  If you would like to control this instead, [PAPERLESS_OCR_USER_ARGS](https://docs.paperless-ngx.com/configuration/#PAPERLESS_OCR_USER_ARGS) exists for basically this thing.

**Comment by shamoon** [comment]
There is no "fingerpointing", this project is FOSS which relies on other free tools, the issue you are describing is not widespread and we do not have the resources to be chasing down every random PDF OCR / rendering issue. As you've noticed, PDF is a very buggy format. So no, we're not going to change the _default_ renderer of the highly-used OCR software we use based off 1 user.

<img width="246" alt="Screenshot 2023-11-03 at 1 32 18 AM" src="https://github.com/paperless-ngx/paperless-ngx/assets/4887959/9c150a45-c94b-4c37-a4c3-51431fd75066">

stumpylog offers a great solution for you

**Comment by r2ixuz** [comment]
Sorry guys, don't wanted to sound harsh yesterday. I know you do this in your free time. And I am very thankful that you do.

For now I'm using PAPERLESS_OCR_USER_ARGS as a workaround.

In my understanding this problem is for every user, not just me. @shamoon your screenshot let me think that you do not understand that issue. Selecting multiple words do not show the issue. You need to double click on a single word to show the wrong selection (even it is correctly fully selected). But the most annoying part is when you try to select manually a long word and you don't know how far you have to select OVER the end of an word (basically you need to select spaces after the word) to fully select this word.

**Comment by shamoon** [comment]
This is built-in viewer

<img width="313" alt="Screenshot 2023-11-03 at 2 20 15 AM" src="https://github.com/paperless-ngx/paperless-ngx/assets/4887959/ae459634-2d21-4618-8f3b-7bc210be50e7">

https://github.com/paperless-ngx/paperless-ngx/assets/4887959/7e41ea77-a52e-46a6-84f6-d9ed4507dee3

**Comment by r2ixuz** [comment]
Well than I don't know. I tried everything: Different client PDF viewer (Adobe Reader, Chrome Browser), different client OS (Linux, Windows, macOS).

So it must be the PDFs produced by my scanner (Brother ADS-2400N) and/or that I set the OCR language to deu (German).

I guess I have to live with the workaround. All I could find is that HOCR renderer is only good for latin alphabet. Should be no problem for me.
